### DISCUSSION TITLE
Namespacing for ROS1

### DISCUSSION THREAD

**Discussion by Herakles100** [novatel/novatel_oem7_driver#53] [Q&A]
Hello, I have a NovAtel PwrPak7D and NovAtel oem7 ros driver setup on one of my computers. I am trying to integrate this into an autonomous stack. I would really like the option to have modular namespaces so that I can run the novatel nodes under the robot's namespace (for example: robot/novatel/oem7/... ). This is particularly useful because I would then have the ability to run multiple robots or gps/imu units. Currently, I am not able to make this change because the namespaces are hardcoded for ROS1 and any change that I tried had not worked (theoretically, I believe that I would just have to remove the prepending slashes). How do I add this functionality into the driver?

**Comment by Timple** [comment]
There was a decision made not to fix this. I've opened #15 for this. 

Decision here: https://github.com/novatel/novatel_oem7_driver/pull/46#discussion_r824267492

**Reply by Herakles100** [reply]
I understand the decision but is there anything that I can do to work around this issue? I kinda get not wanting this change to preserve backwards compatibility of the driver but is there a local fix that would work for you/me?
